spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/ComposedTaskNode.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.core.dsl;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.cloud.dataflow.core.dsl.graph.Graph;
+
+/**
+ * The root AST node for any AST parsed from a composed task specification.
+ *
+ * Andy Clement
+ */
+public class ComposedTaskNode extends AstNode {
+
+	/**
+	 * The DSL text that was parsed to create this ComposedTaskNode.
+	 */
+	private String composedTaskText;
+
+	/**
+	 * The sequence of LabelledNodes parsed from the specification.
+	 */
+	private List<LabelledComposedTaskNode> sequences;
+
+	/**
+	 * All the apps mentioned in the composed task definition.
+	 */
+	private Set<String> taskApps;
+
+	public ComposedTaskNode(String composedTaskSpecification, List<LabelledComposedTaskNode> sequences) {
+		super((sequences.size()==0) ? 0 : sequences.get(0).getStartPos(), 
+				(sequences.size()==0) ? 0 : sequences.get(sequences.size()-1).getEndPos());
+		this.composedTaskText = composedTaskSpecification;
+		this.sequences = sequences;
+	}
+
+	@Override
+	public String stringify(boolean includePositionInfo) {
+		StringBuilder s = new StringBuilder();
+		if (sequences != null) {
+			for (int i=0;i<sequences.size();i++) {
+				if (i>0) {
+					s.append("\n");
+				}
+				s.append(sequences.get(i).stringify(includePositionInfo));
+			}
+		}
+		return s.toString();
+	}
+
+	/**
+	 * @return all the apps referenced in the composed task.
+	 */
+	public Set<String> getTaskApps() {
+		if (taskApps == null) {
+			TaskAppsCollector collector = new TaskAppsCollector();
+			accept(collector);
+			taskApps = collector.getTaskApps();
+		}
+		return taskApps;
+	}
+	
+	public void accept(ComposedTaskVisitor visitor) {
+		visitor.startVisit(this.composedTaskText);
+		int sequenceNumber = 0;
+		for (LabelledComposedTaskNode ctn: sequences) {
+			if (visitor.preVisitSequence(ctn, sequenceNumber)) {
+				ctn.accept(visitor);
+				visitor.postVisitSequence(ctn, sequenceNumber);
+			}
+			sequenceNumber++;
+		}
+		visitor.endVisit();
+	}
+	
+	/**
+	 * @return this AST converted to a Graph form for display by Flo
```

Comment:
This is another consistency issue? I would normally add a period to the main body text of javadoc but not to the param/return strings. looking across the codebase as a whole I don't see periods on the end of param/return strings.